flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/LocalStandaloneFlinkResource.java

```diff
@@ -51,16 +58,28 @@
 
 	private static final Logger LOG = LoggerFactory.getLogger(LocalStandaloneFlinkResource.class);
 
-	private final FlinkDistribution distribution = new FlinkDistribution();
+	private final TemporaryFolder temporaryFolder = new TemporaryFolder();
+	private final Path distributionDirectory;
+	@Nullable
+	private final Path logBackupDirectory;
 	private final FlinkResourceSetup setup;
 
-	LocalStandaloneFlinkResource(FlinkResourceSetup setup) {
+	private FlinkDistribution distribution;
+
+	LocalStandaloneFlinkResource(Path distributionDirectory, Optional<Path> logBackupDirectory, FlinkResourceSetup setup) {
+		this.distributionDirectory = distributionDirectory;
+		this.logBackupDirectory = logBackupDirectory.orElse(null);
 		this.setup = setup;
 	}
 
 	@Override
 	public void before() throws Exception {
-		distribution.before();
+		temporaryFolder.create();
+		Path tmp = temporaryFolder.newFolder().toPath();
+		LOG.info("Copying distribution to {}.", tmp);
+		TestUtils.copyDirectory(distributionDirectory, tmp);
+
+		distribution = new FlinkDistribution(tmp);
```

Comment:
`distribution = createFlinkDistribution();`

Reply:
meh, seems unnecessary. The whole method is about setting up the distribution; it seems arbitrary to move some of that into another method.